### DISCUSSION TITLE
rotate_local_z reset

### DISCUSSION THREAD

**Discussion by ChanManChan** [bevyengine/bevy#12179] [Q&A] [answered]
![image](https://github.com/bevyengine/bevy/assets/42992965/8127ad46-90f6-44b6-8038-7f931ee7b46a)

I'm trying to tilt the ship slightly when turning left and right. But as soon a the ship point towards the bottom of the screen it flips completely even though I have added the -current_z to the roll. I don't understand why this is happening. I'm still learning bevy.

https://github.com/bevyengine/bevy/assets/42992965/6fff6c47-d2f2-41a2-aea7-2290475c450e

**Comment by tim-blackbird** [comment, marked as answer]
`transform.rotation.z` is not the rotation around the Z axis.
To get the current rotation around Z you need `transform.rotation.to_euler(EulerRot::XYZ).2`.

I hope this helps :)

**Comment by ChanManChan** [comment]
Thanks, your suggestion helped but now I get this other issue where the asset just completely flips 180 facing down.

https://github.com/bevyengine/bevy/assets/42992965/44027521-f94b-4987-9894-3d961ae8ab19

Also is there a better way to reset the rotate_local_z value to default?
